agent/06_agent_create_cluster.sh

```diff
@@ -23,9 +23,65 @@ function create_pxe_files() {
 function create_image() {
     local asset_dir=${1}
     local openshift_install=${2}
+
+    if [ "${AGENT_USE_APPLIANCE_MODEL}" == true ]; then
+      create_factory_image
+    else
+      create_automated_image
+    fi
+}
+
+function create_automated_image() {
     "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create image
 }
 
+function create_factory_image() {
+    config_image_drive="sdd"
+
+    # The command to create the config-image must be run out of a separate asset directory using same assets
+    mkdir -p ${config_image_dir}
+    if [[ ${AGENT_USE_ZTP_MANIFESTS} == true ]]; then
+       mkdir -p ${config_image_dir}/cluster-manifests
+       cp ${asset_dir}/cluster-manifests/*.yaml ${config_image_dir}/cluster-manifests/
+    else
+       cp ${asset_dir}/*.yaml ${config_image_dir}
+    fi
+
+    # Create the unconfigured ignition and include it in an ISO
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create cluster-manifests
+
+    # Remove any static networking configuration from unconfigured so that config-image sets it
+    rm "${asset_dir}/cluster-manifests/nmstateconfig.yaml"
+    rm "${asset_dir}/.openshift_install_state.json"
+
+    "${openshift_install}" --dir="${asset_dir}" --log-level=debug agent create unconfigured-ignition
+    base_iso_url=$(oc adm release info --registry-config "$PULL_SECRET_FILE" --image-for=machine-os-images --insecure=true $OPENSHIFT_RELEASE_IMAGE)
+    oc image extract --path /coreos/coreos-x86_64.iso:$HOME/.cache/agent/image_cache --registry-config "$PULL_SECRET_FILE" --confirm $base_iso_url
```

Comment:
This could break for ARM deployments (arch hard-coded to x86_64)?

Reply:
Good catch. Changed to use uname